libraries/botframework-connector/src/auth/jwtTokenProviderFactoryInterface.ts

```diff
@@ -0,0 +1,21 @@
+/**
+ * @module botframework-connector
+ */
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { DefaultAzureCredential } from '@azure/identity';
+
+/*
+ * A factory that can create OAuth token providers for generating JWT auth tokens.
+ */
+export interface JwtTokenProviderFactoryInterface {

```

Comment:
@joshgummersall should we instead export a same-name interface from jwtTokenProviderFactory.ts instead of having this `JwtTokenProviderFactoryInterface` and have TypeScript [merge](https://www.typescriptlang.org/docs/handbook/declaration-merging.html#merging-interfaces) it for consumers instead?

I realize my aversion is primarily due to the name, but still...

Sub-optimal example in our code base: [TurnContext](https://github.com/microsoft/botbuilder-js/blob/eb58d15b67405fe11a9bcdf958ff5a241feced97/libraries/botbuilder-core/src/turnContext.ts#L137-L146)

Reply:
Also, the access level for this interface does not match [.NET](https://github.com/microsoft/botbuilder-dotnet/pull/5829/files#diff-49d5a695d7c466c9b4a44b89aae393b51fec3905e77be373ae5f99f7944177c9R12), which is public. To match .NET, the interface needs to be exported from `src/auth/index.ts`.

Reply:
I’m not a fan of merging identifiers, it’s not obvious or explicit enough in my opinion. If anything, I think the name should be `IJwtTokenProviderFactory` as we have used that pattern elsewhere in the codebase.

Reply:
Fixed in last two commits I pushed.